src/renderer/atlas/AtlasEngine.cpp

```diff
@@ -400,32 +294,44 @@ try
         }
 
         _api.bufferLineColumn.emplace_back(column);
+    }
 
-        const BufferLineMetadata metadata{ _api.currentColor, _api.flags };
-        FAIL_FAST_IF(column > _api.bufferLineMetadata.size());
-        std::fill_n(_api.bufferLineMetadata.data() + x, column - x, metadata);
+    {
+        std::fill(_api.colorsForeground.begin() + x, _api.colorsForeground.begin() + column, _api.currentForeground);
     }
 
-    _api.lastPaintBufferLineCoord = { x, y };
-    _api.bufferLineWasHyperlinked = false;
+    {
+        const auto shift = _api.lineRendition >= LineRendition::DoubleWidth ? 1 : 0;
+        const auto backgroundRow = _p.backgroundBitmap.begin() + _p.backgroundBitmapStride * y;
+        auto it = backgroundRow + x;
+        const auto end = backgroundRow + (static_cast<size_t>(column) << shift);
+        const auto bg = u32ColorPremultiply(_api.currentBackground);
 
+        for (; it != end; ++it)
+        {
+            if (*it != bg)
+            {
+                _p.backgroundBitmapGeneration.bump();
+                std::fill(it, end, bg);
+                break;

```

Comment:
`backgroundBitmapGeneration` allows us to skip uploading the background color bitmap, if the background color didn't change.